tools/proto2jsonschema/postgen.js

```diff
@@ -120,17 +126,55 @@ function getId(filepath) {
 }
 
 /**
- * Gets the CloudEvent#type represented in the JSON schema.
+ * Gets the cloudevent package represented in the JSON schema.
  * @param {string} filepath The input file path
  * @example filepath: /Documents/github/googleapis/google-cloudevents/jsonschema/google/events/cloud/audit/v1/LogEntryData.json
+ * @returns {string} The CloudEvent package for this file represents.
  * @example out: google.events.cloud.audit.v1
- * @returns {string} The CloudEvent ID for this file represents.
  */
-function getCloudEventType(filepath) {
+function getCloudEventPackage(filepath) {
   const removePrefix = filepath.split('jsonschema/')[1];
   const removeSuffix = removePrefix.substring(0, removePrefix.lastIndexOf("/"));
   return removeSuffix.replace(/\//g, '.');
 }
 
+/**
+ * Gets the CloudEvent properties from the corresponding `events.proto` file.
+ * @param {string} packageName The package name of the CloudEvent.
+ * @return {object} cloudevent The CloudEvent properties object.
+ * @return {string[]} cloudevent.types The CloudEvent type strings.
+ * @return {string} cloudevent.product The CloudEvent product.
+ */
+function getCloudEventProperties(packageName) {
+  const protoPath = packageName.split('.').join('/');
+  const eventPath = path.resolve(`${__dirname}/../../proto/${protoPath}/events.proto`);
+  const proto = protobufjs.loadSync(eventPath);
+  const protoAsJSON = proto.toJSON();
+
+  // Flatten the protobuf representation and look directly for these keys:
+  const CLOUD_EVENT_TYPE = '(google.events.cloud_event_type)';
+  const CLOUD_EVENT_PRODUCT = '(google.events.cloud_event_product)';
+  const flattenedProtoValueMap = flatten(protoAsJSON);
+  const flattenedProtoValueMapEntries = Object.entries(flattenedProtoValueMap);
```

Comment:
Can we not predict the structure of this JSON? It feels a bit odd to just look through every entry in the JSON representation, rather than looking for specific paths (product as a top-level file option, event type as a message option).

I'm okay with it for now, but it doesn't sit quite right.

Reply:
Yeah, I'm not really proud of this solution to extract the cloudevent fields.

With the solution, I wanted to parse the protos (rather than string manipulation).

This is the JSON representation of the proto we get for our events.protos:

```json
{"nested":{"google":{"nested":{"events":{"nested":{"cloud":{"nested":{"audit":{"nested":{"v1":{"options":{"csharp_namespace":"Google.Events.Protobuf.Cloud.Audit.V1","(google.events.cloud_event_product)":"Cloud Audit Logs"},"nested":{"AuditLogWrittenEvent":{"options":{"(google.events.cloud_event_type)":"google.cloud.audit.log.v1.written"},"fields":{"data":{"type":"LogEntryData","id":1}}}}}}}}}}}}}}}
```

Another solution would be to recursively traverse the object and find the key. Since the key is unique, just flattening the map and getting the key sounded fine.

Maybe I should add a comment with an example result from protobufjs's toJSON.